internal/k8s/annotations.go

```diff
@@ -23,7 +23,8 @@ import (
 	"k8s.io/api/networking/v1beta1"
 )
 
-func annotationIsKnown(key string) bool {
+// AnnotationIsKnown checks if an annotation is one Contour knows about.
+func AnnotationIsKnown(key string) bool {
```

Comment:
After seeing this patch, I'm inclined to think that it makes more sense to have the annotation helpers in an `annotation` package, so that the meaning is clearer at the call site.

For example, reading `annotations.ParseUpstreamProtocols`, it is pretty clear to the reader that this is doing something with object annotations, but reading `k8s.ParseUpstreamProtocols` doesn't give any kind of clue, and there's no naming pattern that really helps out.

So, I'd suggest that we
1. pull everything except `ParseTimeout` into an `annotations` package
1. Update the signature of all the export to take a single `metav1.ObjectMetaAccessor` parameter and call `GetAnnotations()` internally
1. Rename export to not stutter, so `k8s.ValidAnnotationForKind` would become `annotations.ValidforKind`.

As commented below, I think `ParseTimeout` is a general util, not closely related to annotations, so the `k8s` package seems OK for that.